internal/providers/terraform/hcl_provider.go

```diff
@@ -539,10 +539,19 @@ func marshalBlock(block *hcl.Block, jsonValues map[string]interface{}) {
 
 		childValues := marshalAttributeValues(key, b.Values())
 		if len(b.Children()) > 0 {
-			marshalBlock(b, childValues)
+			p.marshalBlock(b, childValues)
 		}
 
 		if v, ok := jsonValues[key]; ok {
+			if _, ok := v.(json.RawMessage); ok {
+				p.logger.WithFields(log.Fields{
+					"parent_block": block.LocalName(),
+					"child_block":  b.LocalName(),
+				}).Debugf("skipping attribute: %s that has also been declared as a child block", key)
```

Comment:
```suggestion
				}).Debugf("skipping attribute '%s' that has also been declared as a child block", key)
```
ignore if this is too picky.